pylint/checkers/magic_methods.py

```diff
@@ -0,0 +1,148 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
```

Comment:
I think this should be an optional check in pylint/extensions instead. We try to not add opinionated new check so installing a new version is not too painful.

Reply:
I haven't looked at the actual list yet, but shouldn't we be able to come up with an un-opinionated list of methods? The example of `str()` vs. `__str__()` seems like a non controversial lint message.

Reply:
I've already excluded all the dunder methods that don't have an alternative way of being called so the remaining ones should all be ones with an obvious alternative function/operator.

Reply:
Fair enough !